src/ripple/app/paths/PathRequest.cpp

```diff
@@ -474,25 +474,30 @@ PathRequest::getPathFinder(RippleLineCache::ref cache,
     return currency_map[currency] = std::move(pathfinder);
 }
 
-void
+bool
 PathRequest::findPaths (RippleLineCache::ref cache, int const level,
     Json::Value& jvArray)
 {
-    auto sourceCurrencies = sciSourceCurrencies;
-    if (sourceCurrencies.empty ())
+    std::set<ripple::Issue> sourceCurrencies;
+    if (sciSourceCurrencies.empty ())
     {
-        auto usCurrencies =
-            accountSourceCurrencies(*raSrcAccount, cache, true);
-        bool sameAccount = *raSrcAccount == *raDstAccount;
+        auto usCurrencies = accountSourceCurrencies(*raSrcAccount, cache, true);
+        bool const sameAccount = *raSrcAccount == *raDstAccount;
         for (auto const& c : usCurrencies)
         {
-            if (!sameAccount || (c != saDstAmount.getCurrency()))
+            if (! sameAccount || c != saDstAmount.getCurrency())
             {
+                if (sourceCurrencies.size() >= RPC::Tuning::max_auto_src_cur)
+                    return false;
```

Comment:
Am I reading this correctly that if the `raSrcAccount` has > 88 trustlines, then it'll be forced to specify some subset of them to do any kind of path finding? I imagine that's pretty rare.